src/python/zquantum/core/testing/generate_cases_for_backend_tests.py

```diff
@@ -0,0 +1,333 @@
+import sympy
+from sympy.physics.quantum import TensorProduct
+from operator import mul
+
+# This script is used to generate test cases for tests in `core/interfaces/backend_test.py`
+# In order to use just run this python file and then copy-paste the console output to the appropriate place in `test_cases_for_backend_tests.py`
+
+I = sympy.Matrix([[1, 0], [0, 1]])
+H = sympy.Matrix(
+    [[1 / sympy.sqrt(2), 1 / sympy.sqrt(2)], [1 / sympy.sqrt(2), -1 / sympy.sqrt(2)]]
+)
+X = sympy.Matrix([[0, 1], [1, 0]])
+Y = sympy.Matrix([[0, -sympy.I], [sympy.I, 0]])
+Z = sympy.Matrix([[1, 0], [0, -1]])
+S = sympy.Matrix([[1, 0], [0, sympy.I]])
+T = sympy.Matrix([[1, 0], [0, (1 + sympy.I) / sympy.sqrt(2)]])
+
+CNOT = sympy.Matrix([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 0, 1], [0, 0, 1, 0]])
+CZ = sympy.Matrix([[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, -1]])
+SWAP = sympy.Matrix([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
```

Comment:
This will be handled during rework, but just noting that this might cause "single source of truth" issues for gate definitions

Reply:
This is helper code anyway, so I think we'll figure this out :)